pkg/scheduler/reconciliation/reconciliation.go

```diff
@@ -146,3 +146,29 @@ func concatStateReasons(state model.OperationState, reasons []string) (string, e
 	}
 	return strings.Join(reasons, ", "), nil
 }
+
+func operationAlreadyInState(op *model.OperationEntity, state model.OperationState) error {
+	if op.State == state {
+		return newAlreadyInStateError(op)
+	}
+	return nil
+}
+
+type alreadyInStateError struct {
+	op *model.OperationEntity
+}
+
+func (err *alreadyInStateError) Error() string {
+	return fmt.Sprintf("cannot update state of operation '%s' because it is already in state %s", err.op.Component, err.op.State)
```

Comment:
As we addressed this during call, we are not clear why it's an error, I leave this comment here as a bookmark, to address in our meeting.

Reply:
As already explained, typical Go-Style programming.

Here is a reference for you: https://gabrieltanner.org/blog/golang-error-handling-definitive-guide